exopy_qm/instruments/drivers/QuantumMachine.py

```diff
@@ -112,6 +116,24 @@ def is_paused(self):
     def resume(self):
         self.job.resume()
 
+    def wait_for_pause(self):
+        """waits for the program to be paused
+        """
+        while not self.is_paused():
+            time.sleep(0.01)
+        
+    def iterate(self):
+        """Iterates the program by resuming it and feeding True to the input sting 'iterate' 
+        """
+        self.job.resume()
+        self.job.insert_input_stream('iterate', True)
+
+    def finish(self):
+        """Finishes the program by resuming it and feeding False to the input sting 'iterate'
+          """
+        self.job.resume()
+        self.job.insert_input_stream('iterate', False)
+
```

Comment:
The uses of input streams are in the driver, just above

Reply:
Oups, didn't see that line